src/actions/index.js

```diff
@@ -11,6 +11,23 @@ const createAction = (type, payload) => ({ type, payload })
  * THUNKS SECTION *
  ******************/
 
+export const setCurrentUser = () => async dispatch => {
+    dispatch(createAction(actions.SET_CURRENT_USER))
+
+    try {
+        const { currentUser } = await api.getCurrentUser()
+
+        dispatch(createAction(actions.SET_CURRENT_USER_SUCCESS, currentUser))
```

Comment:
I've seen this pattern in the usage-analytics app as well. It seems a bit uncommon to me. It's a bit inbetween writing an object literal and dispatching that, and using an action creator.

Personally for readability and consistency I think I'd be in favor of action creators. I'd find that easier to parse as it's more common, and it'd allow for easier reuse of actions. It's recommended in the redux docs as well: https://redux.js.org/tutorials/fundamentals/part-7-standard-patterns.

However, I can imagine that that's out of scope for this PR if this pattern has already been established. Just wanted to mention it.

Reply:
Thanks, I'll bear this in mind for the next time I work on a redux app. I can't remember where I picked up the pattern I am currently using, but you are right I've used it in several apps. 

I agree this is out-of-scope for this PR and actually think that this does not need to be fixed. At some point we need to refactor this app to use declarative data fetching and at that stage we'll probably get rid of redux all together.

Reply:
Sounds good 👍